python-frontend/src/test/java/org/sonar/python/tree/PythonTreeMakerTest.java

```diff
@@ -2809,6 +2809,26 @@ void except_group_multiple() {
     assertThat(((Name) exceptClause2.exception()).name()).isEqualTo("ValueError");
   }
 
+  @Test
+  void typeAliasStatement() {
+    setRootRule(PythonGrammar.TYPE_ALIAS_STMT);
+
+    var astNode = p.parse("type A[B] = str");
+    var statementWithSeparator = new StatementWithSeparator(astNode, null);
+    var typeAlias = treeMaker.typeAliasStatement(statementWithSeparator);
+
+    assertThat(typeAlias).isNotNull();
+    assertThat(typeAlias.typeKeyword()).isNotNull();
+    assertThat(typeAlias.typeKeyword().value()).isEqualTo("type");
+    assertThat(typeAlias.name()).isNotNull();
+    assertThat(typeAlias.name().name()).isEqualTo("A");
+    assertThat(typeAlias.typeParams()).isNotNull();
```

Comment:
Would it be worth checking that the type parameters are indeed what they were assigned? Or is this test case handled somewhere else?